File: robocode2-server/src/main/java/dev/robocode/tankroyale/server/mappers/BotsToBotsWithIdMapper.java
package dev.robocode.tankroyale.server.mappers;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import dev.robocode.tankroyale.server.model.Bot;
import net.robocode2.schema.BotStateWithId;

public final class BotsToBotsWithIdMapper {

	private BotsToBotsWithIdMapper() {}

	public static List<BotStateWithId> map(Set<Bot> bots) {
		List<BotStateWithId> botStates = new ArrayList<>();
		for (Bot bot : bots) {
			botStates.add(BotToBotStateWithIdMapper.map(bot));
		}
		return botStates;
	}
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/mapper/BotsToBotsWithIdMapper.kt
package dev.robocode.tankroyale.server.mapper

import dev.robocode.tankroyale.schema.BotStateWithId
import dev.robocode.tankroyale.server.model.Bot

object BotsToBotsWithIdMapper {
    fun map(bots: Set<Bot>): List<BotStateWithId> {
        val botStates = mutableListOf<BotStateWithId>()
        bots.forEach { botStates += BotToBotStateWithIdMapper.map(it) }
        return botStates
    }
}